src/modules/mod.ts

```diff
@@ -20,4 +23,30 @@ export class ModModule extends Module {
 			`${msg.author} We don't do job posts here; see <#${rulesChannelId}>`,
 		);
 	}
+
+	@listener({ event: 'guildMemberAdd' })
+	async onJoin(member: GuildMember) {
+		const now = Date.now();
+		const lowerBound = now - RAID_SECONDS * 1000;
+
+		this.joins.push(member);
+
+		// Clean up joins from from outside of our raid threshold.
+		while ((this.joins[0].joinedTimestamp ?? 0) < lowerBound) {
+			this.joins.splice(0, 1);
+			this.bannedUpTo--;
```

Comment:
Probably goes negative a lot while we're not being raided.
```suggestion
			this.bannedUpTo = Math.max(this.bannedUpTo - 1, 0);
```